src/main.rs

```diff
@@ -97,15 +120,40 @@ fn main() {
         )
         .get_matches();
 
+    println!("Auto Selecting Baud Rate for GPS...");
     let port = matches.value_of("port").unwrap();
-    let baud: u32 = matches
-        .value_of("baud")
-        .unwrap_or("9600")
-        .parse()
-        .expect("Could not parse baudrate as an integer");
+    let port_name = port;
+    let mut max_byte_read = 0;
+    let mut correct_baud_rate = 0;
+    for &baud_rate in &BAUD_RATES {
+        match try_baud_rate(port_name, baud_rate) {
+            Ok(byte_read) => {
+                if byte_read > max_byte_read {
+                    max_byte_read = byte_read;
+                    correct_baud_rate = baud_rate;
+                }
+            }
+            Err(_) => {
+                println!("Baud rate {} is not working", baud_rate);
+                continue;
+            }
+        }
+    }
+    if correct_baud_rate == 0 {
+        correct_baud_rate = 38400;
+        println!(
+            "ERROR: No Suitable Baud Rate can be found, setting baud rate to {correct_baud_rate}"
+        );
+    } else if correct_baud_rate != 0 {
+        println!(
+            "Baud rate {} is working on port: {}",
+            correct_baud_rate, port_name
+        );
+        println!("Read {} out of 128 bytes", max_byte_read);
+    }
 
     let s = serialport::SerialPortSettings {
-        baud_rate: baud,
+        baud_rate: correct_baud_rate,
```

Comment:
This is correct, configures the serial port with the detected baud rate, _see next comment_.